chcemvediet/apps/anonymization/models.py

```diff
@@ -96,7 +96,7 @@ def save(self, *args, **kwargs):
         if self.pk is None:  # Creating a new object
             if self.created is None:
                 self.created = utc_now()
-            if self.file:
+            if self.file._file:
                 self.file.name = random_string(10)
                 self.size = self.file.size
```

Comment:
Toto si si tiez zjavne neotestoval, pretoze `self.file` je vzdy false, kym subor nie je savnuty. Bez ohladu na to ci ako parameter pre konstruktor sa pouzil `file=ContentFile('aaa')` alebo `file=None`. Samotna hodnota pre file je ulozena v `self.file._file`.

Sam som o tom nevedel, ale preto sa kod pred commitnutim ma pretestovat. Keby si si to pretestoval, vsimol by si si, ze sa ti ziadne subory do `media/attachment_normalizations/` neukladaju. A ze `size` je stale `NULL`.

Reply:
`Toto je nazorny sposob akym som to testoval. Nepouzival som ContentFile ale File. Vytvorili sa nove subory do media/attachment_normalizations/. Skontroloval som si databazu v admine aj priamo vytvoreny subor na disku. Ak sme ako parameter file zvolili None, ulozili sa <FieldFile: None>. Fungovalo to tak som si myslel ze to je spravne.


\# A -> Attachment
\# AN -> AttachmentNormalization
>> a320 = A.objects.get(id=320)

>>> f = open('/home/vilo/Desktop/dev/normalized.pdf', 'rb')
>>> ff = File(f)
>>> an_new = AN.objects.create(attachment=a320, file=ff, content_type="application/pdf", successful=True)
>>> an_new
<AttachmentNormalization: 43>
>>> bool(an_new.file)
True

>>> an_new = AN.objects.create(attachment=a320, file=File(a320.file), content_type="text_plain", successful=True)
>>> an_new
<AttachmentNormalization: 41>
>>> bool(an_new.file)
True


>>> an_new = AN.objects.create(attachment=a320, file=None, content_type="application/pdf", successful=False)
>>> an_new
<AttachmentNormalization: 44>
>>> an_new.file
<FieldFile: None>
>>> an_new.file is None 
False
>>> bool(an_new.file)
False
`

Reply:
@viliambalaz Aha, chapem, to dava zmysel, kedze `File` ma defaultne nastaveny nazov suboru, z ktoreho sa nacital, tak `an_new.file` sa vyhodnoti ako true, kedze Django to vyhodnocuje podla toho, ci subor ma nastavneme meno. `ContentFile` nema nastavene ziadne meno, tak to vyhodnoti ako false. Asi je to bug v Djangu samotnom, ze zle vyhodnocuju true/false hodnotu pre `an_new.file`.

V kode pouzivame `ContentFile` lebo aj ked berieme data z nejakeho suboru na disku, tak pouzivat `File` mi pride nebezpecne, lebo imho si nevieme byt isty, ze omylom takto nieco s tym suborom na disku spravime. Bezpecnejsie je ten subor precitat a potom robit iba s obsahom suboru.